### ISSUE TITLE
FR: move the WiFi connection status popup to the status bar

### ISSUE THREAD

**Issue by LeslieFH** [koreader/koreader#11098]
Currently, when I turn on my Kobo H2O, I get a WiFi connection status window in the middle of the screen, blocking some of the book. I can start reading, but frequently I have to stop and wait for the WiFi connection to resolve itself, which takes some time and is slightly annoying.

If the popup was to display above status bar in the bottom of the screen, the entire page would be available for reading during the wifi connection.

**Comment by NiLuJe** [comment]
Wouldn't help much, it's a blocking operation on Kobo anyway.

If you actually use Wi-Fi on resume pretty often, you'll have much better luck enabling "Restore Wi-Fi on resume", as that'll happen in the background instead.

**Comment by Justalurker1213** [comment]
Turning on "restore Wi-Fi on resume" still has the connecting box in the middle of the screen every time I wake my kobo from sleep. Is there another way to make it happen in the background?

**Comment by NiLuJe** [comment]
No, and it's unrelated to that. If you see a popup on resume, it's because you have something that requires Wi-Fi running in the background (at a guess, I would say an auto sync from KOSync).

And you won't be able to get away without it if you actually need that sync. I think my comment above might predate the KOSync/WiFi rework, so ignore it.

The TL;DR is:

* If something needs Wi-Fi, you get to see it happen.
* "Restore Wi-Fi on resume" *is* completely silent, but doesn't help with any of the above (and won't, ever).

**Comment by dv-blk** [comment]
Any way to move it? 

I personally don't mind it being a blocking operation - I can usually read the page and it will be done before I finish

**Comment by NiLuJe** [comment]
Not really. It can possibly be a tiny bit too verbose for a Notification (instead of the currently used InfoWIdget), but that might be something worth trying out.

**Comment by nrgbistro** [comment]
I just started using KOReader and this is my biggest issue so far. Like @dv-blk, I don't really mind that it blocks touch input, but since it blocks the center of the page I often have to wait until it completes for me to continue reading. Would love an option to offset this notification to the bottom of the screen.

**Comment by giejay** [comment]
Same for me. I'm even fine with making it a background thread if that could mean missing 1 page of syncing. Would be nice if this would be user configurable

**Comment by milktoastrat** [comment]
I love koreader, its just crazy to me that this is still a thing. Now that syncing between devices is so common, I think it's time to implement a way for koreader to connect to wifi in the background so it isnt impeding on the experience of using such a great app.

**Comment by emes81** [comment]
> No, and it's unrelated to that. If you see a popup on resume, it's because you have something that requires Wi-Fi running in the background (at a guess, I would say an auto sync from KOSync).
> 
> And you won't be able to get away without it if you actually need that sync. 

https://github.com/brugsbells/Koreader-Patches

I installed the 2-disable-wifi-notifications.lua patch from here on my Clara BW. I then enabled auto progress sync, which also forces wifi to connect. Now, wifi connects automatically - without the blocking pop-up. It's a major usability win.

I haven't yet tested this away from a wifi connection that the Clara BW knows, but I'll try that and report back. 

In any case, it seems like there's at least the germ of something really useful in that patch. It would be great to see something from it integrated into KOreader itself, so that we can have KOsync without the blocking popup.

**Comment by NiLuJe** [comment]
Oh, it's still blocking, you're just no longer seeing anything *telling you* why the application isn't responding to input ;).

**Comment by NiLuJe** [comment]
> Wouldn't help much, it's a blocking operation on Kobo anyway.
> 
> If you actually use Wi-Fi on resume pretty often, you'll have much better experience enabling "Restore Wi-Fi on resume", as that'll happen in the background instead.

^ Your best best is still this (... on platforms where this is supported ;p).

**Comment by ELLIOTTCABLE** [comment]
Added my :+1: up top, this is one of my biggest frustrations at the moment, besides some actual syncing issues with the underlying sync service I'm using. (It's made worse by my experience with #15106, but that's slightly unrelated - just definitely exacerbates the annoyance of this issue.)

I read that you're saying this isn't possible, but it's not clear to me *why* it's not possible; so let me narrow it down with two specific questions that may clarify things:

1. *What* is being blocked, when you say it's a blocking operation?

    - Do you mean the background operation that triggered the connection can't complete until it's connected? Because that's obvious, and fine - right now, I don't care if the progress-sync takes a moment when it powers on; 80% of the time, the 'last time I was reading' was seven seconds ago when I turned the device off to set it down for a second, and I'm turning it on to continue reading the exact page that is currently already on the e-ink screen.
    - Do you mean user-input, like buttonpresses or touches, are dropped and lost? If so, yes, that sucks, but "not being able to change the page" (a mildly common activity) is still less bad than "not being able to read the page already on screen (a literally constant, primary-purpose-of-the-device activity.)

2. Even if it is some fundamental technical issue, I'm still with @nrgbistro, and don't understand why that's not the behaviour - even if it's absolutely critical for some reason I'm missing that the user *know* the network-activity isn't completed yet, why is it necessary for that to block the content of the page?

In particular, I want to draw parallels to the "reflow" operation that shows up as a tiny icon in the top-left. It stays out of your way, even though when it's complete, all the text on screen may completely change and disappear! It's, yes, unavoidable, that post-reflow, your reading may be interrupted ... but nonetheless, in *some* cases, the reflow *doesn't* negatively affect your reading-experience; and although in the worst-cases, it can be surprising/confusing for the text to suddenly jump around and you lose your place ... it's still more important that in the *majority* of cases, there's not a popup/modal suddenly blocking your reading up until that point.

So, my vote: since we already *have* that convention, now, my vote is that we double-down on it. Blocking/breaking/limitation situations that affect the user's core ability to read the page in front of them and have basic expectations such as "i can read and continue to read and then hit the page turn button and continue to read" break, should *always*, consistently, result in a warning-indicator in the top corner.

(If possible, tapping the top left corner during that period should expand into details about *why* - and if the answer to num.1 above is "yes indeed user-input is dropped", then replace the 'connecting' icon with a 'connection successful' icon once control is restored and keep it on screen for a little while, so the user still has something they can *eventually* tap to ask "wtf just happened why wasn't my device responding and what does this icon mean.")

So, whatever the technical limitations and negative effect on the user-experience, I really don't believe it's *worse* than, every few minutes that I'm using the device, it becomes completely non-functional for its purpose (... reading/referencing the page displayed on the face of the device) while some network stuff figures itself out in the background. (=

**Comment by claptraw** [comment]
I agree with the previous speakers. 

I installed KOReader on my Kobo Libra 2 a few days ago and using abs-kosync-bridge to sync my reading progress between KOReader, Booklore and Audiobookshelf. That’s why I always need an active Wi-Fi connection. 

However, the OS (or KOReader?) often seems to turn off Wi-Fi, even when I’ve unchecked “Disable Wi-Fi connection when inactive.” As I understand it, disabling that option should make Wi-Fi stay on at all times - but it doesn’t. And when the reader has been in standby for a while, you regularly get those annoying Wi-Fi connection banners. 

I would really appreciate an option to completely disable these intrusive banners AND, at the same time, allow touch inputs while the connection is being established. Or simply keep Wi-Fi permanently enabled.

**Comment by emes81** [comment]
> Oh, it's still blocking, you're just no longer seeing anything _telling you_ why the application isn't responding to input ;).

i 

> > Wouldn't help much, it's a blocking operation on Kobo anyway.
> > If you actually use Wi-Fi on resume pretty often, you'll have much better experience enabling "Restore Wi-Fi on resume", as that'll happen in the background instead.
> 
> ^ Your best best is still this (... on platforms where this is supported ;p).

I think what was actually happening was that I had that setting enabled, which is why there was no blocking pop-up.

How come it's possible to have non-blocking wifi connection occur in the background on resume, but not in other contexts?

**Comment by emes81** [comment]
> Wouldn't help much, it's a blocking operation on Kobo anyway.
> 
> If you actually use Wi-Fi on resume pretty often, you'll have much better experience enabling "Restore Wi-Fi on resume", as that'll happen in the background instead.

It would help in one way: removing the huge blocker from the middle of the page would mean it would still be possible to keep on reading while waiting for the connection to resolve

**Comment by shuvashish76** [comment]
As workaround you could assign 2 actions to a gesture and set it as "Execute" then you won't see any such WiFi connection status window. I've setup "Toggle Wi-Fi" and "Full screen refresh" to a single gesture.
If screen refresh is distracting for you then you can cleverly setup other action in addition with "Toggle Wi-Fi" such as "Reading statistics: enable" _(it's won't affect reading-statistics as it triggers for enable as always)_

IDK if it's a bug but I'm using it as a feature 🫣
@Frenzie is there a reason it behaves this way when using multiple actions as single gesture? intentional?

**Comment by Frenzie** [comment]
I'm not sure if I understood, but I seem to recall you were using Android? It's completely asynchronous there.

**Comment by shuvashish76** [comment]
Tested on Kobo Clara BW no such WiFi connection status window if combined with another action for same gesture. No notification no distractions. _(Only checked with connecting existing wifi)_

**Comment by Frenzie** [comment]
That's probably unintentional but it doesn't sound like it matters. ;-)

**Comment by shurane** [comment]
I think https://github.com/koreader/koreader/issues/14527 is related.

**Comment by shurane** [comment]
I guess the mechanism for turning on the wifi or keepalive is way different on my Kindle Paperwhite 11th gen vs the Kobo Clara Color. 

Kindle Paperwhite is instant. Maybe it's always connected? Battery never drops though.

Kobo Clara Color takes 10+ seconds every time, it's most annoying when it goes idle and then I have to wake it up again. I wasn't able to recreate it this time so maybe it does keep the Wifi connection up for a bit before turning it off. Both are running `2026.07.1` (or whatever the latest version that's pulled via the Update button on the menu).

https://github.com/user-attachments/assets/e81bbbe6-7a24-4a2c-8220-6aaf180c51c0
